sdk/storage/storage-file-datalake/src/models.ts

```diff
@@ -939,6 +939,179 @@ export interface FileReadToBufferOptions extends CommonOptions {
   concurrency?: number;
 }
 
+/**
+ * Base type for options to query file.
+ *
+ * @export
+ * @interface FileQueryTextConfiguration
+ */
+export interface FileQueryTextConfiguration {}
+
+/**
+ * Options to query file with JSON format.
+ *
+ * @export
+ * @interface FileQueryJsonTextConfiguration
+ */
+export interface FileQueryJsonTextConfiguration extends FileQueryTextConfiguration {
+  /**
+   * Record separator.
+   *
+   * @type {string}
+   * @memberof FileQueryJsonTextConfiguration
+   */
+  recordSeparator: string;
+  /**
+   * Query for a JSON format file.
+   *
+   * @type {"json"}
+   * @memberof FileQueryJsonTextConfiguration
+   */
+  kind: "json";
+}
+
+/**
+ * Options to query file with CSV format.
+ *
+ * @export
+ * @interface FileQueryCsvTextConfiguration
+ */
+export interface FileQueryCsvTextConfiguration extends FileQueryTextConfiguration {
+  /**
+   * Record separator.
+   *
+   * @type {string}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  recordSeparator: string;
+  /**
+   * Query for a CSV format file.
+   *
+   * @type {"csv"}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  kind: "csv";
+  /**
+   * Column separator. Default is ",".
+   *
+   * @type {string}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  columnSeparator?: string;
+  /**
+   * Field quote.
+   *
+   * @type {string}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  fieldQuote?: string;
+  /**
+   * Escape character.
+   *
+   * @type {string}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  escapeCharacter?: string;
+  /**
+   * Has headers. Default is false.
+   *
+   * @type {boolean}
+   * @memberof FileQueryCsvTextConfiguration
+   */
+  hasHeaders?: boolean;
+}
+
+/**
+ * File query error type.
+ *
+ * @export
+ * @interface FileQueryError
+ */
+export interface FileQueryError {
+  /**
+   * Whether error is fatal. Fatal error will stop query.
+   *
+   * @type {boolean}
+   * @memberof FileQueryError
+   */
+  isFatal: boolean;
+  /**
+   * Error name.
+   *
+   * @type {string}
+   * @memberof FileQueryError
+   */
+  name: string;
+  /**
+   * Position in bytes of the query.
+   *
+   * @type {number}
+   * @memberof FileQueryError
+   */
+  position: number;
+  /**
+   * Error description.
+   *
+   * @type {string}
+   * @memberof FileQueryError
+   */
+  description: string;
+}
+
+/**
+ * Option interface for Data Lake file - query operations
+ *
+ * See:
+ * - {@link DataLakeFileClient.query}
+ *
+ * @export
+ * @interface FileQueryOptions
+ */
+export interface FileQueryOptions extends CommonOptions {
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
```

Comment:
Is the `&commat;` intentional?

Reply:
I think so. Checked the generated API doc, it looks fine.
`For example, use the @azure/abort-controller to create...`
https://docs.microsoft.com/en-us/javascript/api/@azure/storage-blob/appendblobappendblockoptions?view=azure-node-latest#abortsignal